src/sender.js

```diff
@@ -178,32 +348,74 @@ class Sender {
     }
 
     /**
-     * Closes the connection to the database. <br>
+     * @ignore
+     * @return {http.Agent} Returns the default http agent.
+     */
+    getDefaultHttpAgent() {
+        if (!Sender.DEFAULT_HTTP_AGENT) {
+            Sender.DEFAULT_HTTP_AGENT = new http.Agent(DEFAULT_HTTP_AGENT_CONFIG);
+        }
+        return Sender.DEFAULT_HTTP_AGENT;
+    }
+
+    /**
+     * @ignore
+     * @return {https.Agent} Returns the default https agent.
+     */
+    getDefaultHttpsAgent() {
+        if (!Sender.DEFAULT_HTTPS_AGENT) {
+            Sender.DEFAULT_HTTPS_AGENT = new https.Agent(DEFAULT_HTTP_AGENT_CONFIG);
+        }
+        return Sender.DEFAULT_HTTPS_AGENT;
+    }
+
+    /**
+     * Closes the TCP connection to the database. <br>
      * Data sitting in the Sender's buffer will be lost unless flush() is called before close().
      */
     async close() {
-        const address = this.socket.remoteAddress;
-        const port = this.socket.remotePort;
-        this.socket.destroy();
-        this.log('info', `Connection to ${address}:${port} is closed`);
+        Sender.numOfSenders--;
```

Comment:
Shouldn't we do this only for HTTP senders?

Reply:
Opted for a global counter, as it is a great help if you have to track down leaked senders, but I see your point.
If someone mixes HTTP and ILP senders, the HTTP agents will not get destroyed until all senders are closed, including the TCP ones.
Will separate the counters.

Reply:
done